xwiki-platform-core/xwiki-platform-display/xwiki-platform-display-api/src/main/java/org/xwiki/display/internal/AbstractDocumentTitleDisplayer.java

```diff
@@ -136,9 +141,15 @@ public XDOM display(DocumentModelBridge document, DocumentDisplayerParameters pa
     private XDOM displayTitle(DocumentModelBridge document, DocumentDisplayerParameters parameters)
     {
         // 1. Try to use the title provided by the user.
-        if (!StringUtils.isEmpty(document.getTitle())) {
+        String rawTitle = document.getTitle();
+        if (!StringUtils.isEmpty(rawTitle)) {
             try {
-                return parseTitle(evaluateTitle(document.getTitle(), document.getDocumentReference(), parameters));
+                String title = rawTitle;
+                // Evaluate the title only if the document has script rights, otherwise use the raw title.
+                if (authorizationManager.hasAccess(Right.SCRIPT)) {
```

Comment:
The general architecture is to check for security at the boundaries of the XWiki system and once in, to not check again for perf reasons. So I think I agree with Marius.